src/EncryptedPayload.ts

```diff
@@ -50,6 +50,27 @@ class EncryptedPayload {
         json: string,
     ): EncryptedPayload {
 
+        const jsonObject = EncryptedPayload.validateJson(json);
+
+        // validate obligatory fields
+        if (!jsonObject.hasOwnProperty('ciphertext')) {
```

Comment:
I guess this would be the hacky bit. To cope with the existing Opt in paper wallet we don't have much choice??
@ AdriaCarrera @fboucquez ??